packages/editor/src/browser/semantic-highlight/semantic-highlighting-service.ts

```diff
@@ -0,0 +1,46 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Position, Range } from 'vscode-languageserver-types';
+import URI from '@theia/core/lib/common/uri';
+import { Disposable } from '@theia/core/lib/common/disposable';
+
+export const SemanticHighlightingService = Symbol('SemanticHighlightingService');
+export interface SemanticHighlightingService extends Disposable {
+
+    decorate(uri: URI, ranges: SemanticHighlightingRange[]): Promise<void>;
+
+}
+
+@injectable()
+export class NoopSemanticHighlightingService implements SemanticHighlightingService {
```

Comment:
How about turning `SemanticHighlightingService` into a class and use it as no-op implementation to get rid of additional symbol, inteface and bindings? Similary to `QuickOpenService`.